tfhe/src/integer/ciphertext/mod.rs

```diff
@@ -112,6 +156,22 @@ impl IntegerCiphertext for RadixCiphertext {
     }
 }
 
+impl RadixIntegerCiphertext for RadixCiphertext {}
```

Comment:
The trait hierarchies maybe need a bit of a rework ?

Like which one has the blocks and why is block_carries_are_empty separate ?

Reply:
IntegerCiphertext  is the more general thing, we expect the integer to only be a collection of shortints, this works for radix and CRT.

RadixIntegerCiphertext, is less general and adds the notion of carries which radix ciphertext have but not CRT ones

Reply:
May I suggest putting the more general part in first part maybe ?

IntegerCiphertext
IntegerRadixCiphertext

?

Reply:
renamed

Reply:
and understood for the logic for the carries and stuff